### PR TITLE
Add access token support

### PR DESCRIPTION
**Pull request by kamilturek**
This PR adds a support for access token authentication.

https://dev.targetprocess.com/docs/authentication#personal-access-tokens-set-within-user-interface

### CODE REVIEW

**Comment by kamilturek** [comment]
Hey @OwenPattison, any chance of getting this reviewed? Many thanks!

**Comment by petevb** [comment]
@kamilturek only just seen this sorry. Will pick it up on Monday. Thanks for the PR!

File: index.ts [hunk 1]
```diff
@@ -10,19 +11,29 @@ enum APIVersion {
...
 
-    constructor(subdomain: string, username: string, password: string) {
+    constructor(
+        subdomain: string,
+        username?: string,
+        password?: string,
+        config?: { accessToken?: string }
+    ) {
         this.subdomain = subdomain;
+        this.accessToken = config && config.accessToken;
```
Comment (petevb) [inline index.ts:26]: Please use Elvis operator.
```suggestion
        this.accessToken = config?.accessToken;
```

**Review by petevb — COMMENTED** [review]
Tiny nit, otherwise LGTM. Thanks!

**Review by petevb — APPROVED** [review]
:shipit:

**Comment by petevb** [comment]
linter didn't like that - reverting.

File: index.ts [hunk 2]
```diff
@@ -10,19 +11,29 @@ enum APIVersion {
...
 
-    constructor(subdomain: string, username: string, password: string) {
+    constructor(
+        subdomain: string,
+        username?: string,
+        password?: string,
+        config?: { accessToken?: string }
+    ) {
         this.subdomain = subdomain;
+        this.accessToken = config?.accessToken;
```
Comment (petevb) [inline index.ts:26]:
```suggestion
        this.accessToken = config && config.accessToken;
```

**Comment by kamilturek** [comment]
@petevb Thanks for the review. I think that the reason your suggestion didn't work is that the optional chaining [was introduced in Typescript 3.7](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) and this package uses `3.0.3`.

https://github.com/NewOrbit/targetprocess-rest-api/blob/fe51913c4d080ac18a7f4145ec29ecc1a0c8d919/package.json#L40

I didn't want to bump up the version on my own though.

**Comment by petevb** [comment]
> @petevb Thanks for the review. I think that the reason your suggestion didn't work is that the optional chaining [was introduced in TypeScript 3.7](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) and this package uses `3.0.3`.
> 
> https://github.com/NewOrbit/targetprocess-rest-api/blob/fe51913c4d080ac18a7f4145ec29ecc1a0c8d919/package.json#L40
> 
> I didn't want to bump up the version on my own though.

Makes sense, thank you for coming back to me and pointing that out - I assumed it was something like that but didn't have time to investigate.  Thanks again for the PR, @kamilturek! 🥇

**Pull request merged**